unlock-app/src/components/interface/locks/Settings/forms/ReceiptBaseForm.tsx

```diff
@@ -0,0 +1,131 @@
+import { useMutation, useQuery } from '@tanstack/react-query'
+import { TextBox } from '@unlock-protocol/ui'
+import { Input, Button } from '@unlock-protocol/ui'
+import { useForm } from 'react-hook-form'
+import { ToastHelper } from '~/components/helpers/toast.helper'
+import { storage } from '~/config/storage'
+
+import { SupplierBodyProps } from '../../../../../../../locksmith/src/controllers/v2/receiptBaseController'
+
+interface ReceiptBaseFormProps {
+  lockAddress: string
+  network: number
+  isManager: boolean
+  disabled: boolean
+}
+export const ReceiptBaseForm = ({
+  lockAddress,
+  isManager,
+  disabled,
+  network,
+}: ReceiptBaseFormProps) => {
+  const {
+    register,
+    handleSubmit,
+    getValues,
+    reset,
+    formState: { isValid },
+  } = useForm<SupplierBodyProps>({
+    mode: 'onChange',
+  })
+
+  const getReceiptBase = async (): Promise<any> => {
+    return await storage.getReceiptsBase(network, lockAddress)
+  }
+
+  const saveReceiptBase = async (): Promise<any> => {
+    if (!isManager) return
+    const supplier = await getValues()
+    return await storage.saveReceiptsBase(network, lockAddress, {
+      data: {
+        ...supplier,
+      },
+    })
+  }
+
+  const receiptBaseMutation = useMutation(saveReceiptBase)
+
+  const onHandleSubmit = async () => {
+    if (isValid) {
+      await ToastHelper.promise(receiptBaseMutation.mutateAsync(), {
+        loading: 'Updating  supplier details,',
+        success: 'Supplier updated',
+        error: 'We could not update the details.',
+      })
+    } else {
+      ToastHelper.error('Form is not valid')
+      reset()
+    }
+  }
+
+  const { isLoading: isLoadingDetails } = useQuery(
+    ['getReceiptBase', lockAddress, network, receiptBaseMutation.isSuccess],
+    async () => {
+      return getReceiptBase()
+    },
+    {
+      onSuccess: (res) => {
+        if (res.data) {
+          reset(res.data)
+        }
+      },
+      enabled: isManager,
+    }
+  )
+
```

Comment:
Could extract these into hooks similar to what I did for metadata? Makes components much easier to read and reuse.